src/ring_theory/integral_closure.lean

```diff
@@ -710,6 +711,36 @@ begin
   rw [mul_assoc, ← pow_succ', tsub_add_cancel_of_le this]
 end
 
+lemma is_field_of_is_integral_of_is_field'
+  {R S : Type*} [comm_ring R] [is_domain R] [comm_ring S] [is_domain S] [algebra R S]
+  (H : algebra.is_integral R S) (hR : is_field R) :
+  is_field S :=
+begin
+  letI := hR.to_field R,
+  refine ⟨⟨0, 1, zero_ne_one⟩, mul_comm, λ x hx, _⟩,
+  let A := algebra.adjoin R ({x} : set S),
+  let f : A →ₗ[R] A :=
+  { to_fun := λ y, ⟨x, algebra.mem_adjoin_iff.mpr (subring.mem_closure.mpr (λ T hT,
+      show _, from set.singleton_subset_iff.mp (set.union_subset_iff.mp hT).2))⟩ * y,
```

Comment:
Can't you get here much faster with `subset_adjoin? I'd expect you can also use `algebra.lmul` to avoid needing to prove map_add and map_smul.

Reply:
Ah, thanks for pointing out `subset_adjoin` and `algebra.lmul`. I thought those things would exist, but had an awful lot of trouble finding them!